src/vue/common/CollectionLoader.spec.js

```diff
@@ -0,0 +1,114 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import CollectionLoader from './CollectionLoader'
+import { globalize } from '../filters/globalize'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+const localVue = createLocalVue()
+localVue.filter('globalize', globalize)
+let wrapper
+const someData = []
+const requestPerPage = 10
+const someEvent = 'some-event'
+
+beforeEach(() => {
+  wrapper = shallowMount(CollectionLoader, {
+    propsData: {
+      firstPageLoader: sinon.stub().resolves({ data: someData }),
+      requestPerPage
+    },
+    localVue
+  })
+})
+afterEach(() => {
+  sinon.restore()
+})
+describe('CollectionLoader component test', () => {
+  it('loadPage correctly load loader fn', async () => {
+    const spy = sinon.stub().resolves({ data: someData })
+
+    await wrapper.vm.loadPage(someEvent, spy)
+
+    expect(spy.calledOnce).to.be.true
+  })
+
+  it('loadPage correctly emit event with data', async () => {
+    await wrapper.vm.loadPage(someEvent, sinon.stub().resolves({
+      data: someData
+    }))
+
+    expect(wrapper.emitted()[someEvent]).to.exist
+    expect(wrapper.emitted()[someEvent].length).to.equal(1)
+    expect(wrapper.emitted()[someEvent][0][0]).to.equal(someData)
```

Comment:
also there is another one such assertion in the project. pls write a helper that will do all the stuff in one row

Reply:
ok